signup.js

```diff
@@ -0,0 +1,103 @@
+
+const signupEmailInput = document.querySelector('#signup__email-form');
+const signupPwInput = document.querySelector('#signup__password-form');
+const signupPwCheckInput = document.querySelector('#signup__pwscheck-form');
+
+const signupEmailError = document.querySelector('.signup__email-form--error');
+const signupPwError = document.querySelector('.signup__password-form--error');
+const signupPwCheckError = document.querySelector('.signup__pwcheck-form--error');
+
+const signupButton = document.querySelector('.signup__form--signupBt');
+const signupPasswordIcon = document.querySelector('.password__toggle-icon');
+const signupPwcheckIcon = document.querySelector('.pwcheck__toggle-icon');
+
+signupEmailInput.addEventListener('focusout', signupEmailFocusOut);
+signupPwInput.addEventListener('focusout', signuppwFocusOut);
+signupPwCheckInput.addEventListener('focusout', signupPwCheckFocusOut);
+
+signupPasswordIcon.addEventListener('click', passwordToggleIcon);
+signupPwcheckIcon.addEventListener('click', pwcheckToggleIcon)
+signupButton.addEventListener('click', handlesignupButtonClick);
+
+signupEmailInput.addEventListener('keyup', handlesignupButtonEnter);
+signupPwInput.addEventListener('keyup', handlesignupButtonEnter);
+signupPwCheckInput.addEventListener('keyup', handlesignupButtonEnter);
+
+
+//이메일 입력 확인 (공통)
+function signupEmailFocusOut(event) {
+  input.emailHandleInputFocusOut(signupEmailInput, signupEmailError);
+}
+
+//비밀번호 확인 (공통)
+function signuppwFocusOut(event) {
+  input.passwordHandleFocusOut(signupPwInput, signupPwError);
+}
+
+//입력한 비밀번호 재확인 
+function signupPwCheckFocusOut(event) {
+  pwCheckHandleFocusOut(signupPwCheckInput, signupPwInput, signupPwCheckError);
+}
+
+function pwCheckHandleFocusOut(inputElement, inputcompareElement, errorElement) {
+  if (inputcompareElement.value != inputElement.value) {
+    errorElement.textContent = '비밀번호가 일치하지 않습니다.';
+    errorElement.style.display = 'block';
+    inputElement.style.borderColor = '#FF5B56';
+  } else {
+    input.noError(inputElement, errorElement);
+  }
+}
+
+//비밀번호 아이콘 및 타입 변경 (공통)
+function passwordToggleIcon(event) {
+  input.togglePasswordType(signupPwInput, signupPasswordIcon);
+}
+
+function pwcheckToggleIcon(event) {
+  input.togglePasswordType(signupPwCheckInput, signupPwcheckIcon);
+}
+
+//엔터키
+function handlesignupButtonEnter(event) {
+  if (event.code === 'Enter') {
+    handlesignupButtonClick();
+  }
+}
+
+//로그인 버튼 클릭 시
+function handlesignupButtonClick(event) {
+  if(signupEmailInput.value !== '' && signupEmailInput.value !== '' && signupEmailInput.value !== '' ){
+    if (signupEmailInput.value !== 'test@codeit.com') {
+      if (signupPwInput.value === signupPwCheckInput.value) {
+        window.location.href = "/folder";
+      }
+      else {
+        //비밀번호 일치하지않음
+        pwCheckHandleFocusOut(signupPwCheckInput, signupPwInput, signupPwCheckError);
+      }
+    }
+    else {
+      emailExistence();
+    }
+  } else {
+    blinkInput();
+  }
+}
```

Comment:
if문이 중첩되어 있으면 좋지 않습니다.
실무에서 웬만하면 if문은 1번만 쓰고 어쩔 수 없는경우에 2번정도 중첩해서 씁니다. if문을 줄일려면 지금 사용하신 조건문을 뒤집고 return하면 됩니다. 

if문을 하나씩만 쓰면서 대신에 아래로 내려가면서 데이터를 필터링한다고 생각하시면 됩니다.


```suggestion
if (
        signupEmailInput.value === "" ||
        signupEmailInput.value === "" ||
        signupEmailInput.value === ""
    ) {
        blinkInput();
        return;
    }

    if (signupEmailInput.value !== "test@codeit.com") {
        emailExistence();
        return;
    }

    if (signupPwInput.value === signupPwCheckInput.value) {
        window.location.href = "/folder";
    } else {
        //비밀번호 일치하지않음
        pwCheckHandleFocusOut(signupPwCheckInput, signupPwInput, signupPwCheckError);
    }}
```